src/components/ec/cuda/kernel/ec_cuda_reduce.cu

```diff
@@ -212,118 +64,28 @@ __global__ void UCC_REDUCE_CUDA_MULTI_DST_SUM<float>(
     }
 }
 
-#define LAUNCH_KERNEL_A(NAME, type, _AlphaType, _task, s, b, t)                \
+#define LAUNCH_REDUCE_A(NAME, type, _AlphaType, _task, s, b, t)                \
     do {                                                                       \
         if (_task->task_type == UCC_EE_EXECUTOR_TASK_REDUCE) {                 \
-            UCC_REDUCE_CUDA_DEFAULT_##NAME<type, _AlphaType>                   \
+            UCC_REDUCE_CUDA_DEFAULT_##NAME<type, _AlphaType, false>            \
                 <<<b, t, 0, s>>>(_task->reduce, _task->flags);                 \
         } else if (_task->task_type == UCC_EE_EXECUTOR_TASK_REDUCE_STRIDED) {  \
-            ucc_eee_task_reduce_strided_t *trs = &_task->reduce_strided;       \
-            UCC_REDUCE_CUDA_STRIDED_##NAME<type, _AlphaType><<<b, t, 0, s>>>(  \
-                (type *)trs->src1, (type *)trs->src2, (type *)trs->dst,        \
-                trs->count, trs->stride, trs->n_src2,                          \
-                (bool)(_task->flags & UCC_EEE_TASK_FLAG_REDUCE_WITH_ALPHA),    \
-                trs->alpha);                                                   \
+            UCC_REDUCE_CUDA_STRIDED_##NAME<type, _AlphaType, false>            \
+                <<<b, t, 0, s>>>(_task->reduce_strided, _task->flags);         \
         } else {                                                               \
-            UCC_REDUCE_CUDA_MULTI_DST_##NAME<type>                             \
+            UCC_REDUCE_CUDA_MULTI_DST_##NAME<type, false>                      \
                 <<<b, t, 0, s>>>(_task->reduce_multi_dst);                     \
         }                                                                      \
     } while (0)
 
 #define LAUNCH_KERNEL(NAME, type,  _task, s, b, t)  \
```

Comment:
where it's used?

Reply:
In `CUDA_REDUCE_WITH_OP_MULTI_DST` (def in `ec_cuda_reduce_ops.h`) we need to know if we are in the triggered or non-triggered case to define `start` and `step`

Reply:
it is not. removed